### PR TITLE
update libp2p to v0.32.2

### PR DESCRIPTION
**Pull request by envestcc**

### CODE REVIEW

File: p2p.go [hunk 1]
```diff
@@ -328,22 +331,27 @@ func NewHost(ctx context.Context, options ...Option) (*Host, error) {
...
-		libp2p.ConnectionManager(connmgr.NewConnManager(cfg.ConnLowWater, cfg.ConnHighWater, cfg.ConnGracePeriod)),
+		libp2p.ConnectionManager(connMgr),
+		libp2p.DefaultPrometheusRegisterer,
 	}
 
 	if !cfg.SecureIO {
 		opts = append(opts, libp2p.NoSecurity)
 	} else {
-		opts = append(opts, libp2p.Security(secio.ID, secio.New))
+		// opts = append(opts, libp2p.Security(secio.ID, secio.New))
```
Comment (dustinxie) [inline p2p.go:349]: this is not needed?

File: p2p.go [hunk 2]
```diff
@@ -328,22 +331,27 @@ func NewHost(ctx context.Context, options ...Option) (*Host, error) {
...
 	}
 
 	// relay option
 	if cfg.Relay == "active" {
-		opts = append(opts, libp2p.EnableRelay(relay.OptActive, relay.OptHop))
+		opts = append(opts, libp2p.EnableRelay())
```
Comment (dustinxie) [inline p2p.go:354]: could you confirm

File: p2p.go (hunk 1 above)
Comment (envestcc) [reply p2p.go:349]: [secio](https://github.com/libp2p/go-libp2p-secio) is already deprecated:
> secio is not maintained any more, and we advise against using it.
Use [go-libp2p-tls](https://github.com/libp2p/go-libp2p-tls/) and [go-libp2p-noise](https://github.com/libp2p/go-libp2p-noise/) instead.

and tls and noise is the two default security of libp2p, so we can just remove it

File: p2p.go [hunk 3]
```diff
@@ -634,7 +642,7 @@ func (h *Host) AddBootstrap(bootNodeAddrs []multiaddr.Multiaddr) error {
 }
 
 // HostIdentity returns the host identity string
-func (h *Host) HostIdentity() string { return h.host.ID().Pretty() }
+func (h *Host) HostIdentity() string { return h.host.ID().String() }
```
[the remark this answers is not in the document]
Comment (envestcc) [reply p2p.go:645]: yes, they are all base58-encoded

File: p2p.go (hunk 2 above)
Comment (envestcc) [reply p2p.go:354]: there is no options any more in the new version.

**Review by dustinxie — APPROVED** [review]

**Pull request merged**